mira/metamodel/composition.py

```diff
@@ -0,0 +1,257 @@
+        return hash(self.author.name)
+
+    def __eq__(self, other):
+        if isinstance(other, AuthorWrapper):
+            return self.author.name == other.author.name
+        return False
+
+
+def compose(tm_list):
+    """
+    Compose a list of template models into a single template model
+
+    Parameters
+    ----------
+    tm_list :
+        The list of template models to compose
+
+    Returns
+    -------
+    :
+        The composed template model derived from the list of template models
+    """
+    if len(tm_list) < 2:
+        raise ValueError(f"Expected the list of template models to be at "
+                         f"least length 2.")
+
+    intermediate_results = [_compose_two_tms(tm_list[0], tm_list[1])]
+
+    for intermediate_index, tm in enumerate(tm_list[2:]):
+        intermediate_results.append(_compose_two_tms(
+            intermediate_results[intermediate_index], tm)
+        )
+
+    return intermediate_results[-1]
+
+
+def _compose_two_tms(tm0, tm1):
+    """
+    Helper method that composes two template models into one
+
+    Parameters
+    ----------
+    tm0 :
+        The first template model to be composed
+    tm1 :
+        The second template model to be composed
+
+    Returns
+    -------
+    :
+        The composed template model
+    """
+    # what would rate law conflict look like, and where would it appear
+    # Would it appear in the case that the similarity score is 1.0 between
+    # two template models but the corresponding templates from the two
+    # models have a different rate-laws?
+    # Do we define model priority order by the specificity of the concepts
+    # in the templates in question? Or can it be a user-defined parameter?
+    model_list = [tm0, tm1]
+    rf_func = get_dkg_refinement_closure().is_ontological_child
+    compare = TemplateModelComparison(model_list,
+                                      refinement_func=rf_func)
+    comparison_result = compare.model_comparison.get_similarity_scores()
+
+    if comparison_result[0]["score"] == 0:
+        # get the union of all template model attributes
+        # as the models are 100% distinct
+        new_templates = tm0.templates + tm1.templates
+        new_parameters = {**tm0.parameters, **tm1.parameters}
+        new_initials = {**tm0.initials, **tm1.initials}
+        new_observables = {**tm0.observables, **tm1.observables}
+        new_annotations = annotation_composition(tm0.annotations,
+                                                 tm1.annotations)
+
+        return TemplateModel(templates=new_templates,
+                             parameters=new_parameters,
+                             initials=new_initials,
+                             observables=new_observables,
+                             annotations=new_annotations)
+
+    elif comparison_result[0]['score'] == 1.0:
+        # return the first template model as both
+        # template models are exactly the same
+        return tm0
```

Comment:
This might be fine but it's also probably just a special case of the else branch here